kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/consumer/KafkaConsumerService.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.homeoffice.digital.sas.kafka.consumer;
+
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_CONSUMING_MESSAGE;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.homeoffice.digital.sas.kafka.message.KafkaEventMessage;
+import uk.gov.homeoffice.digital.sas.kafka.validators.SchemaValidator;
+
+@Slf4j
+@Service
+@Getter
+public abstract class KafkaConsumerService<T> {
+
+  protected KafkaEventMessage<T> kafkaEventMessage;
+
+  private ObjectMapper mapper = new ObjectMapper();
+
+  @Value("${kafka.resource.name}")
+  private String resourceName;
+
+  @Value("${kafka.supported.schema.version}")
+  private DefaultArtifactVersion supportedVersion;
+
+  private final SchemaValidator schemaValidator;
+
+  protected KafkaConsumerService(SchemaValidator schemaValidator) {
+    this.schemaValidator = schemaValidator;
+  }
+
+  public KafkaEventMessage<T> consume(String payload
+  ) throws JsonProcessingException {
+    schemaValidator.setSupportedVersion(supportedVersion);
+    schemaValidator.setResourceName(resourceName);
```

Comment:
As `supportedVersion` and `resourceName` are only used by SchemaValidator, why not inject them directly in SchemaValidator bean?